utils/watchcat/files/watchcat.init

```diff
@@ -90,53 +90,36 @@ load_watchcat() {
 		logger -p user.err -t "watchcat" "reboot program $1 not started - $error"
 		return
 	}
+	
+	procd_open_instance
```

Comment:
I wouldn't do it that way. If the switch case is not [defined](https://github.com/openwrt/packages/pull/14569/files#diff-34a04232b2d4ace6c3a9fc39702b0c51cc79a7a00825329430c10674d1faf700L108), then the object was opened anyway. I don't think that's clean.
I think it would be better to save the command for procd_set_param and then pass it to a complete procd_* block. And if the mode in switch case is not found then to a return to proceed to the next mode

Reply:
> I think it would be better to save the command for procd_set_param and then pass it to a complete procd_* block. And if the mode in switch case is not found then to a return to proceed to the next mode

Agreed, there is nothing stopping it from opening an instance when the input is invalid, so it needs to be done that way.